src/spicy/spicyz/main.cc

```diff
@@ -250,6 +273,21 @@ int main(int argc, char** argv) try {
         return 1;
     }
 
+    // Spicy uses two special strings to generate identifiers, some of which
+    // are weak symbols which need to be unique across the whole program, in
+    // particular the HLTO scope. Spicy parsers compiled into HLTO files
+    // register themself with the Spicy runtime, but we also inject code so
+    // they register themself as analyzers, both of which happens in the
+    // context of a HLTO scope. In order to support compiling the same parsers
+    // into different analyzer HLTOs we need to make the weak HLTO scopes
+    // unique symbols, so we generate unique strings based on inputs here to
+    // support that use case.
```

Comment:
That this needs such a long comment is a strong smell that this doesn't do things exactly right. The alternative would be to push this into e.g., Spicy's `Driver::parseOptions`, but AFAICT Spicy doesn't really need this.

Reply:
I'll take a look at this.